aries_cloudagent/protocols/present_proof/v2_0/formats/dif/handler.py

```diff
@@ -0,0 +1,268 @@
+
+    async def create_exchange_for_request(
+        self,
+        pres_ex_record: V20PresExRecord,
+        pres_request_message: V20PresRequest,
+    ) -> None:
+        """Create a presentation exchange record for input presentation request."""
+
+    async def create_bound_request(
+        self,
+        pres_ex_record: V20PresExRecord,
+        request_data: dict = None,
+    ) -> Tuple[V20PresFormat, AttachDecorator]:
+        """
+        Create a presentation request bound to a proposal.
+
+        Args:
+            pres_ex_record: Presentation exchange record for which
+                to create presentation request
+            name: name to use in presentation request (None for default)
+            version: version to use in presentation request (None for default)
+            nonce: nonce to use in presentation request (None to generate)
+            comment: Optional human-readable comment pertaining to request creation
+
+        Returns:
+            A tuple (updated presentation exchange record, presentation request message)
+
+        """
+        dif_proof_request = V20PresProposal.deserialize(
+            pres_ex_record.pres_proposal
+        ).attachment(self.format)
+
+        return self.get_format_data(PRES_20_REQUEST, dif_proof_request)
+
+    async def create_pres(
+        self,
+        pres_ex_record: V20PresExRecord,
+        request_data: dict = {},
+    ) -> Tuple[V20PresFormat, AttachDecorator]:
+        """Create a presentation."""
+        proof_request = V20PresRequest.deserialize(
+            pres_ex_record.pres_request
+        ).attachment(self.format)
+        pres_definition = None
+
+        if request_data != {}:
+            pres_spec_payload = DIFPresSpecSpecSchema().load(request_data)
+            # Overriding with prover provided pres_spec
+            challenge = str(uuid4())
+            domain = None
+            pres_definition = pres_spec_payload.get("presentation_definition")
+            issuer_id = pres_spec_payload.get("issuer_id")
+        if not pres_definition:
+            challenge = proof_request.get("challenge") or str(uuid4())
+            domain = proof_request.get("domain") or None
+            pres_definition = PresentationDefinition.deserialize(
+                proof_request.get("presentation_definition")
+            )
+            issuer_id = None
+
+        input_descriptors = pres_definition.input_descriptors
+        try:
+            holder = self._profile.inject(VCHolder)
+            types = []
+            schema_ids = []
+            for input_descriptor in input_descriptors:
+                for schema in input_descriptor.schemas:
+                    uri = schema.uri
+                    required = schema.required or True
+                    if required:
+                        # JSONLD Expanded URLs
+                        if "#" in uri:
+                            types.append(uri.split("#")[1])
```

Comment:
I think it is wrong to assume the schema.uri is a type if it contains a `#`. Take a look at this playground: https://tinyurl.com/ye9whcca 

the expanded types result in:
```
    "@type": [
      "https://www.w3.org/2018/credentials#VerifiableCredential",
      "http://schema.org/Person"
    ]
```

So a `type` CAN include a `#`, but this is not always the case

Reply:
I think the most 'correct' approach would be to store the expanded types and make an overall search for the schema.uri's where either the `@context` or the `@type` is the value. I don't think there is an easy way to determine whether it is a context or type

Reply:
I think this is what I have done in the latest update.

Reply:
Updated: Saving expanded_types [ `f"xpnd:type:{expanded_type_val}": "1"` ] in `cred_tags` of a `VCRecord`. Inside `present_proof/v2_0/pres_exch_handler`, first try to match/search `schema.uri` with `VCRecord` `schema_ids`, if not found, then match/search `f"xpnd:type:{schema.uri}": "1"` against `cred_tags`.